packages/headless/src/api/knowledge/knowledge-api-answer.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {GeneratedContentFormat} from '../../features/generated-answer/generated-response-format';
+import {selectQuery} from '../../features/query/query-selectors';
+import {QueryState} from '../../features/query/query-state';
+import {selectSearchHub} from '../../features/search-hub/search-hub-selectors';
+import {ConfigurationSection} from '../../state/state-sections';
+import {GeneratedAnswerCitation} from '../generated-answer/generated-answer-event-payload';
+import {SearchRequest} from '../search/search/search-request';
+import {knowledgeApi} from './knowledge-api';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+interface GeneratedAnswerStream {
+  answerStyle: string | undefined;
```

Comment:
Is there no stricter type than string for this?

Reply:
yes :+1: